bluesky/plans.py

```diff
@@ -2078,10 +2086,17 @@ def inner_product_scan(detectors, num, *args, per_step=None, md=None):
     """
     if md is None:
         md = {}
+
+    # mds can't handle ophyd devices, so all replace them with reprs where
+    # necessary:
+    mds_args = [repr(arg) if hasattr(arg, 'name') else arg
```

Comment:
:bike: :house: I would prefer not inferring this from `.name` and instead building `mds_args` as part of the loop over `chunked(args, 3)` below, where we know for sure which pieces are which.